File: pfa4-main/pfa4-main/amn-frontend/src/app/models/patient-profile.dto.ts
import { MedicalRecordDTO } from './medical-record.dto';
import { Vaccination } from './vaccination.dto';
import { VisitLog } from './visit-log.dto';
import { ScanDTO } from './scan.dto';
import { SurgeryDTO } from './surgery.dto';
import { PrescriptionDTO } from './prescription.dto';
import { AnalysisDTO } from './analysis.dto';

export interface PatientProfileDTO {
  id: number;
  fullName: string;
  cin: string;
  bloodType?: string;
  birthDate?: string;
  emergencyContact?: string;
  allergies?: string;
  chronicDiseases?: string;
  hasHeartProblem: boolean;
  hasSurgery: boolean;
  email: string;
  phone: string;
  address?: string;
  medicalFolderId?: number;

  // Medical Records
  medicalRecords: MedicalRecordDTO[];

  // Vaccinations
  vaccinations: Vaccination[];

  // Visit Logs
  visitLogs: VisitLog[];

  // Scans
  scans: ScanDTO[];

  // Surgeries
  surgeries: SurgeryDTO[];

  // Prescriptions
  prescriptions: PrescriptionDTO[];

  // Analyses
  analyses: AnalysisDTO[];
}

File: pfa4-main/pfa4-main/src/main/java/com/amn/dto/PatientProfileDTO.java
package com.amn.dto;

import com.amn.entity.*;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PatientProfileDTO {
    private Long id;
    private String fullName;
    private String cin;
    private String bloodType;
    private String birthDate;
    private String emergencyContact;
    private String allergies;
    private String chronicDiseases;
    private boolean hasHeartProblem;
    private boolean hasSurgery;
    private String email;
    private String phone;
    private String address;
    private Long medicalFolderId;

    // Use raw entity lists — NO DTOs
    private List<MedicalRecordDTO> medicalRecords;

    private List<Vaccination> vaccinations;
    private List<VisitLog> visitLogs;
    private List<ScanDTO> scans;
    private List<SurgeryDTO> surgeries;
    private List<PrescriptionDTO> prescriptions;
    // ✅ Fix here


    private List<AnalysisDTO> analyses;
}
